cli/src/commands/branch.rs

```diff
@@ -251,14 +251,19 @@ fn cmd_branch_create(
         workspace_command.resolve_single_rev(args.revision.as_ref().unwrap_or(&RevisionArg::AT))?;
     let view = workspace_command.repo().view();
     let branch_names = &args.names;
-    if let Some(branch_name) = branch_names
-        .iter()
-        .find(|&name| view.get_local_branch(name).is_present())
-    {
-        return Err(user_error_with_hint(
-            format!("Branch already exists: {branch_name}"),
-            "Use `jj branch set` to update it.",
-        ));
+    for name in branch_names {
+        if view.get_local_branch(name).is_present() {
+            return Err(user_error_with_hint(
+                format!("Branch already exists: {name}"),
+                "Use `jj branch set` to update it.",
+            ));
+        }
+        if has_tracked_remote_branches(view, name) {
+            return Err(user_error_with_hint(
+                format!("Tracked remote branches exist for deleted branch: {name}"),
+                "Use `jj branch set` to restore local branch.",
```

Comment:
Optional:

I think it'd be nicer if this printed `name@remote`, not just `name`.

Also, I think you could add "..., or `jj git push` to delete the branch on remote(s) before trying again" (assuming this is true).

Reply:
> I think it'd be nicer if this printed `name@remote`, not just `name`, if it's easy. Or it could say "... on remotes A, B, C".

I tend to agree, but I wanted to keep the implementation simple. It's unlikely that user has to deal with this situation often.

> Also, I think you could add "..., or `jj git push` to delete the branch on remote(s) before trying again" to the hint (assuming this is true).

Hmm, suppose the user doesn't remember why he deleted tracking local branch, `jj branch forget` (or `untrack`) is a better option. WDYT?

Reply:
The best I could come up with so far is:

> Use `jj branch set` to restore the local branch, `jj branch untrack 'glob:{name}@*'` to forget that this branch existed locally.

I think both the behavior of `jj branch forget` in this situation and the reason that the above instruction works are a bit confusing. I wonder whether we should have `jj forget --local {name}` be an alias for `jj branch untrack 'glob:{name}@*'`. This makes sense especially if we think that the behavior of the two might diverge later.

I'm also wondering whether the first clause should say:

> Use `jj branch s {name} -r {name}@REMOTE`...

Reply:
> The best I could come up with so far is:
> 
> > Use `jj branch set` to restore the local branch, `jj branch untrack 'glob:{name}@*'` to forget that this branch existed locally.

Sounds good, thanks. I would rephrase the latter part as "to untrack remote branches." `"forget"` in this context implies remote-tracking branches will be deleted locally.

> I think both the behavior of `jj branch forget` in this situation and the reason that the above instruction works are a bit confusing. I wonder whether we should have `jj forget --local {name}` be an alias for `jj branch untrack 'glob:{name}@*'`. This makes sense especially if we think that the behavior of the two might diverge later.

Perhaps, it's the matter whether `forget NAME` should delete untracked remotes or not? I don't think the current `jj forget` behavior is correct, but it generally works as expected.

BTW, I don't expect `forget --local {name}` is equivalent to `untrack 'glob:{name}@*'`. The former should delete remote-tracking branches, but the latter shouldn't.

> I'm also wondering whether the first clause should say:
> 
> > Use `jj branch s {name} -r {name}@REMOTE`...

It seems too specific. The user's intent is unclear here. Maybe "to restore the local branch" in the hint is also too specific.

Reply:
I read this late, but I sort-of answered this in https://github.com/martinvonz/jj/pull/3884#issuecomment-2172040123.

I think the current `forget` is awkward and shouldn't be the default. I currently think about calling it `forget --global`, and requiring the user to specify either `--local` or `--global`. In the long term, I'd probably be OK with `--local` becoming the default, but I'm not sure that's necessary.

Reply:
Update hint to suggest both `jj branch set` and `jj branch untrack glob:..`. PTAL.